post-processor/shell-local/post-processor.go

```diff
@@ -68,7 +69,12 @@ func (p *PostProcessor) Configure(raws ...interface{}) error {
 	}
 
 	if p.config.ExecuteCommand == "" {
-		p.config.ExecuteCommand = `chmod +x "{{.Script}}"; {{.Vars}} "{{.Script}}"`
+		if runtime.GOOS == "windows" {
+			// chmod doesn't exist on windows!
+			p.config.ExecuteCommand = `{{.Vars}} {{.Script}}`
```

Comment:
I'm afraid this might be too naive... Just looking at the powershell provisioner, and it's exceedingly complicated (cf setting variables, running the command)

the windows-shell provisioner isn't much less complicated. I think we either need to encapsulate the logic of running powershell/cmd shell so we can reuse it here, or come up with some alternative

Reply:
You're thinking the "p.config.ExecuteCommand = `if (Test-Path variable:global:ProgressPreference){$ProgressPreference='SilentlyContinue'};{{.Vars}}&'{{.Path}}';exit $LastExitCode`"

Do we have a place for reasonable defaults that should be getting reused?